src/test/java/com/virtuslab/archunit/ClassStructureTestSuite.java

```diff
@@ -117,7 +117,7 @@ public void inner_classes_should_not_be_instantiated_from_constructor_of_enclosi
             new DescribedPredicate<JavaConstructorCall>(
                 "an inner class is instantiated in a constructor of the enclosing class") {
               @Override
-              public boolean apply(JavaConstructorCall input) {
```

Comment:
from https://github.com/TNG/ArchUnit/releases/tag/v1.0.0-rc1:
> `DescribedPredicate` now extends the JDK 8 Predicate, so `apply(..)` has to be replaced by `test(..)`

I think it only required change to get it working. However, all the changes and features may be worth reviewing and possibly applied

Reply:
Potentially relevant changes:

> Accesses from inside of lambdas are now correctly detected. Before, the origin was set as a synthetic method like lambda$xxx$123 instead. JavaAccess can now be queried for isDeclaredInLambda() to distinguish this from an access outside of a lambda (see https://github.com/TNG/ArchUnit/pull/847; thanks a lot to @oberprah, @FrederikFinckh)

> New methods codeUnits().should().onlyBeCalled().by{Classes/Methods/...}That(..)

Reply:
Opened https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1212

Reply:
> Accesses from inside of lambdas are now correctly detected

Reply:
ok, i got it

Reply:
somehow it lambdas pass this check:
```java 
if (method.getName().startsWith("access$") || method.getName().startsWith("lambda$")) {
     return;
}
```

Reply:
but we can now use:
```java
if (call.isDeclaredInLambda()) {
    // ignore
}
```

Reply:
To be fixed within #1212